mmedit/models/backbones/video_interpolation_backbones/cain_net.py

```diff
@@ -0,0 +1,343 @@
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            Tensor: Output tensor with shape (n, c, h, w).
+        """
+
+        y = self.body(x)
+        y = self.conv_after_body(y)
+        return x + y
+
+
+@BACKBONES.register_module()
+class CAINNet(nn.Module):
+    """CAIN network structure.
+
+    Paper: Channel Attention Is All You Need for Video Frame Interpolation.
+    Ref repo: https://github.com/myungsub/CAIN
+
+    Args:
+        in_channels (int): Channel number of inputs. Default: 3.
+        out_channels (int): Channel number of outputs. Default: 3.
+        kernel_size (int): Kernel size of CAINNet. Default: 3.
+        num_block_groups (int): Number of block groups. Default: 5.
+        num_block_layers (int): Number of blocks in a group. Default: 12.
+        depth (int): Down scale depth, scale = 2**depth. Default: 3.
+        reduction (int): Channel reduction of CA. Default: 16.
+        norm (str | None): Normalization layer. If it is None, no
+            normalization is performed. Default: None.
+        padding (int): Padding of CAINNet. Default: 7.
+        act (function): activate function. Default: nn.LeakyReLU(0.2, True).
+    """
+
+    def __init__(self,
+                 in_channels=3,
+                 out_channels=3,
+                 kernel_size=3,
+                 num_block_groups=5,
+                 num_block_layers=12,
+                 depth=3,
+                 reduction=16,
+                 norm=None,
+                 padding=7,
+                 act=nn.LeakyReLU(0.2, True)):
+        super().__init__()
+
+        mid_channels = in_channels * pow(4, depth)
+        self.scale = pow(2, depth)
+        self.padding = padding
+
+        self.conv_first = nn.Conv2d(mid_channels * 2, mid_channels, 3, 1, 1)
+        self.body = make_layer(
+            ResidualGroup,
+            num_block_groups,
+            block_layer=ResidualChannelAttention,
+            num_block_layers=num_block_layers,
+            mid_channels=mid_channels,
+            kernel_size=3,
+            reduction=reduction,
+            norm=norm,
+            act=act)
+        self.conv_last = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1)
+
+    def forward(self, imgs, padding_flag=False):
+        """Forward function.
+
+        Args:
+            imgs (Tensor): Input tensor with shape (n, 2, c, h, w).
+            padding_flag (bool): Padding or not. Default: False.
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        assert imgs.shape[1] == 2
+        x1, x2 = imgs[:, 0], imgs[:, 1]
+
+        mean1 = x1.mean(2, keepdim=True).mean(3, keepdim=True)
+        mean2 = x2.mean(2, keepdim=True).mean(3, keepdim=True)
+        x1 -= mean1
+        x2 -= mean2
+
+        if padding_flag:
+            padding_function, depadding_function = get_padding_functions(
+                x1, self.padding)
+            x1 = padding_function(x1)
+            x2 = padding_function(x2)
```

Comment:
Is there a case when `self.padding > 0` but `padding_flag=False`?

Reply:
It will not pad in this case.

Reply:
So can use only `self.padding` to determine? 

Specifically, how is `padding_flag` set? If `padding_flag` is set based on the size of the input, I guess only `self.padding` should be sufficient?

Reply:
`padding_flag` is necessary in #691